### PR TITLE
Update documentation for indexing/slicing methods

### PR DESCRIPTION
**Pull request by citizen428**
See #39911

r? @steveklabnik

### CODE REVIEW

File: src/libcollections/slice.rs [hunk 1]
```diff
@@ -435,7 +441,6 @@ impl<T> [T] {
     }
 
     /// Returns a raw pointer to the slice's buffer.
-    ///
```
Comment (GuillaumeGomez) [inline src/libcollections/slice.rs]: Why this removal?

File: src/libcollections/str.rs [hunk 2]
```diff
@@ -433,7 +433,11 @@ impl str {
     }
 
     /// Creates a string slice from another string slice, bypassing safety
-    /// checks.
+    /// checks. This is generally not recommended, use with caution!
```
Comment (GuillaumeGomez) [inline src/libcollections/str.rs]: Please add an empty line before the new sentence.

File: src/libcollections/str.rs [hunk 3]
```diff
@@ -476,7 +480,11 @@ impl str {
...
```
Comment (GuillaumeGomez) [inline src/libcollections/str.rs]: Please add an empty line before the new sentence.

File: src/libcollections/slice.rs [hunk 4]
```diff
@@ -393,7 +393,10 @@ impl<T> [T] {
     }
 
     /// Returns a reference to an element or subslice, without doing bounds
-    /// checking. So use it very carefully!
+    /// checking. This is generally not recommended, use with caution! For
```
Comment (GuillaumeGomez) [inline src/libcollections/slice.rs]: Please add an empty line before the new sentence.

File: src/libcollections/slice.rs [hunk 5]
```diff
@@ -413,7 +416,10 @@ impl<T> [T] {
     }
 
     /// Returns a mutable reference to an element or subslice, without doing
-    /// bounds checking. So use it very carefully!
+    /// bounds checking. This is generally not recommended, use with caution!
```
Comment (GuillaumeGomez) [inline src/libcollections/slice.rs]: Please add an empty line before the new sentence.

**Comment by GuillaumeGomez** [comment]
All good, just a few nits.

File: src/libcollections/slice.rs (hunk 1 above)
Comment (citizen428) [inline src/libcollections/slice.rs]: Sausage fingers...

**Comment by citizen428** [comment]
@GuillaumeGomez Thanks for the feedback, just updated the PR.

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**